app/app.py

```diff
@@ -28,6 +28,23 @@ def linchpin_init():
         return jsonify(status=409, code=output.returncode)
 
 
+@app.route('/workspace/delete', methods=['POST'])
+def linchpin_delete_workspace():
+    try:
+        data = request.json  # Get request body
+        name = data["name"]
+        # path specifying location of working directory inside server
+        for x in os.listdir(os.path.join(app.root_path)):
+            if x == name:
+                output = subprocess.Popen(["rm", "-r", os.path.join(app.root_path + WORKING_DIR + name)]
```

Comment:
Since we will be heavily using system calls for provisioning and initialization.
It would be preferred to use python library instead of subprocess modules. 
 Quick example for deleting directories goes as follows:
```
# !/usr/bin/python

import os, sys

# listing directories
print "the dir is: %s" %os.listdir(os.getcwd())

# removing path
os.rmdir("mydir")
```

Reply:
Found a thread on how to delete directories : I think this approach would be more efficient 
@mansikulkarni96 
https://stackoverflow.com/questions/48892772/how-to-remove-a-directory-is-os-removedirs-and-os-rmdir-only-used-to-delete-emp

Reply:
Thanks @samvarankashyap